src/web/build_details.rs

```diff
@@ -39,12 +36,13 @@ impl_axum_webpage! {
 }
 
 pub(crate) async fn build_details_handler(
-    Path((name, version, id)): Path<(String, String, String)>,
+    Path((name, version, id)): Path<(String, ReqVersion, String)>,
     mut conn: DbConnection,
     Extension(config): Extension<Arc<Config>>,
     Extension(storage): Extension<Arc<AsyncStorage>>,
 ) -> AxumResult<impl IntoResponse> {
     let id: i32 = id.parse().map_err(|_| AxumNope::BuildNotFound)?;
+    let version = version.assume_exact()?;
```

Comment:
In cases where we're just going to `assume_exact`, could we just take a `Version` in the path instead?

Reply:
valid point, updated the code, removed the method. 

This _will_ change the status code for non-exact version identifiers (404 to 400), but since this matches the change when we have other invalid version identifiers, IMO this is fine.